server/server.js

```diff
@@ -11,11 +16,15 @@ const connection = mongoose.connect(SRV_DB, {
 
 connection
   .then(() => {
-    app.listen(PORT, () =>
-      console.log(`Database connection successful on port ${PORT}`)
-    );
+    app.listen(PORT, () => {
+      createFolderIfNotExists(uploadDir);
```

Comment:
Wyciągłeś nazwy plików i funkcję z tego samego pliku i łączysz je ze sobą w innym. Tu pomimo że jest to tak średnio pomyślane to działa, ale uważaj w takich sytuacjach by nie zapętlić sobie zależności. Ja bym zrobił w tamtym pliku fukncje nazwał ją jakoś initFolders czy coś w tym stylu i tu ją wywołał. No i raczej potrzebujesz sprawdzić 3 rzeczy "tmp", "public" i "public/avatars" bo próba utworzenia avatars jak nie masz public powinna wyrzucić błąd.